src/RA_Dlg_GameLibrary.cpp

```diff
@@ -532,6 +521,110 @@ INT_PTR CALLBACK Dlg_GameLibrary::s_GameLibraryProc(HWND hDlg, UINT uMsg, WPARAM
+    {
+        Expects((hwnd != nullptr) && (pnmhdr != nullptr));
+        switch (idFrom)
+        {
+            case IDC_RA_LBX_GAMELIST:
+            {
+                switch (pnmhdr->code)
+                {
+                    case LVN_ITEMCHANGED:
+                    {
+                        // RA_LOG( "Item Changed\n" );
+                        HWND hList = GetDlgItem(hwnd, IDC_RA_LBX_GAMELIST);
+                        const int nSel = ListView_GetSelectionMark(hList);
+                        if (nSel != -1)
+                        {
+                            TCHAR buffer[1024]{};
+                            ListView_GetItemText(hList, nSel, 1, buffer, 1024);
+                            SetWindowText(GetDlgItem(hwnd, IDC_RA_GLIB_NAME), buffer);
+                        }
+                    }
+                    break;
+
+                    case NM_CLICK:
+                        // RA_LOG( "Click\n" );
+                        break;
+
+                    case NM_DBLCLK:
+                        if (LaunchSelected())
+                        {
+                            EndDialog(hwnd, TRUE);
+                            return TRUE;
+                        }
+                        break;
+
+                    default:
+                        break;
+                }
+            }
+                return FALSE;
+
+            default:
+                RA_LOG("%08x, %08x\n", idFrom, reinterpret_cast<LPARAM>(pnmhdr));
+                return FALSE;
+        }
+    };
+
+    const auto OnCommand = [this](HWND hwnd, int id, HWND hwndCtl, [[maybe_unused]] UINT /*codeNotify*/)
+    {
+        Expects((hwnd != nullptr) && (hwndCtl != nullptr));
+        switch (id)
+        {
+            case IDOK:
+                if (LaunchSelected())
+                {
+                    EndDialog(hwnd, IDOK);
+                    return TRUE;
+                }
+                else
+                {
+                    return FALSE;
+                }
+
+            case IDC_RA_RESCAN:
+            {
+                ReloadGameListData();
+                {
+                    std::scoped_lock lock{mtx};
+                    SetDlgItemText(m_hDialogBox, IDC_RA_SCANNERFOUNDINFO, TEXT("Scanning..."));
+                }
+                return FALSE;
+            }
+
+            case IDC_RA_PICKROMDIR:
+            {
+                std::string sROMDirLocation = GetFolderFromDialog();
+                auto& pConfiguration = ra::services::ServiceLocator::GetMutable<ra::services::IConfiguration>();
+                pConfiguration.SetRomDirectory(sROMDirLocation);
+                RA_LOG("Selected Folder: %s\n", sROMDirLocation.c_str());
+                SetDlgItemText(hwnd, IDC_RA_ROMDIR, NativeStr(sROMDirLocation).c_str());
+                return FALSE;
+            }
+
+            case IDC_RA_LBX_GAMELIST:
+            {
+                const int nSel = ListView_GetSelectionMark(hwndCtl);
```

Comment:
`hwndCtl` is unboxed from an `LPARAM`. But it is the handle to the last control interacted with (usually the same `id` as the case label). In `fallthrough`  cases we couldn't assume that, so left how the handle was retrieved in those cases.